scripts/quests/otherAreas/X_Marks_The_Spot.lua

```diff
@@ -0,0 +1,131 @@
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Despachiaire'] = quest:progressEvent(144),
+
+            onEventFinish =
+            {
+                [144] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            -- Will repeat until the quest is completed.
+            ['Despachiaire'] = quest:progressEvent(144),
+
+            ['Parelbriaux'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') <= 1 then
+                        return quest:progressEvent(145)
+                    end
+                end,
+            },
+
+            ['Odeya'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(141, 0, xi.item.TAVNAZIAN_LIVER)
+                    elseif quest:getVar(player, 'Prog') == 2 then
+                        return quest:progressEvent(146, 0, xi.item.TAVNAZIAN_LIVER)
+                    elseif quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(147)
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(143)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        npcUtil.tradeHasExactly(trade, { { xi.item.TAVNAZIAN_LIVER, 1 } })
+                    then
+                        return quest:progressEvent(142)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [145] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [141] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [142] = function(player, csid, option, npc)
+                        player:confirmTrade()
+                        quest:setVar(player, 'Prog', 3)
+                end,
+
+                [143] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
+            },
+        },
+
+        [xi.zone.PHOMIUNA_AQUEDUCTS] =
+        {
+            ['_0r9'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(37)
+                    else
+                        return
```

Comment:
You can get rid of the else-return here if it's going to do nothing. Presumably the ??? or whatever this is has a DefaultAction set up in the file for that zone?

Reply:
Now that you've changed this and it falls out the bottom, should this 0r9 thing have a default action? What's in the cap?
```
local ID = zones[xi.zone.PHOMIUNA_AQUEDUCTS]

return {
    ['_0r5'] = { messageSpecial = ID.text.DOOR_FIRMLY_SHUT },
}

```

Reply:
In the capture, the door doesn't open and gives a line about the door being firmly shut. I've been doing research to see if there is any instance in which this door opens as it is coded (outside of the cutscene). I was going to put in a capture request to have someone test it again outside of the quest.

Reply:
I've just watched the cap, all the information needed is there. It plays you this message if there isn't a quest involved:
<img width="227" alt="image" src="https://github.com/user-attachments/assets/61f4b6d3-4483-4152-802d-8b5e72e297ea">

So set that as the default action for this door. (there may have been some ID shifts since this cap was made, so figure it out with your local client)

Reply:
I'm suspicious about the doors around here, because they all have this block:
```
    elseif npc:getAnimation() == 9 then
        npc:openDoor()
    end
```

I don't remember the story around this area, if gates are meant to be shut at any time, if you need keys, etc. This might just be copy/paste error, or they might legitimately need to open - and that's hard to figure out from caps

Reply:
I suspect a retail character with large mission completion could quickly go there and see what the door does, but then you'd have to work backwards from "it's normally open, when is it shut?" or "it's normally shut, when is it open?"

Reply:
In terms of this PR then, nothing needs to happen, the underlying object opens like it always has, and you don't change that behaviour 👍

Reply:
I think you're right and it might just be a copy-paste error. @siknoz did a recent capture (almost a year after the x marks the spot capture) for me to double-check the door and it still displays the door is firmly shut. I also couldn't find any instance where the player needs to go into the room beyond here.

![image](https://github.com/user-attachments/assets/04a48f8d-802c-44a5-a0ec-512bc8cb9bf0)


I'm going to go ahead and add in the default action and remove the door opening animation, if it is ever found to have a need then it can be added back.